src/Cardano/Transaction.cpp

```diff
@@ -67,6 +67,26 @@ uint256_t TokenBundle::getAmount(const std::string& key) const {
     return findkey->second.amount;
 }
 
+set<string> TokenBundle::getPolicyIds() const {
+    set<string> policyIds;
+    for (const auto& t: bundle) {
+        if (policyIds.find(t.second.policyId) == policyIds.end()) {
+            policyIds.emplace(t.second.policyId);
+        }
+    }
+    return policyIds;
+}
+
+vector<TokenAmount> TokenBundle::getByPolicyId(const string& policyId) const {
+    vector<TokenAmount> filtered;
+    for (const auto& t: bundle) {
+        if (t.second.policyId == policyId) {
+            filtered.push_back(t.second);
+        }
+    }
+    return filtered;
```

Comment:
seems to be a perfect case for `std::copy_if`

Reply:
It didn't work out with `copy_if`:  only map iterator is possible, while only value is needed to be added.  Maybe transform and then copy_if